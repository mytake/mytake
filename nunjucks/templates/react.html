{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
