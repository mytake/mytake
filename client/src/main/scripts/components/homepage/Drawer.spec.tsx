/*
 * MyTake.org website and tooling.
 * Copyright (C) 2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
import * as React from "react";
import * as renderer from "react-test-renderer";
import Drawer from "./Drawer";
import { INFO_HEADER_TABS_ENUM } from "./infoHeader";

jest.mock("./DrawerContents", () => ({
  __esModule: true,
  default: "DrawerContents",
}));

test("Drawer - expanded", () => {
  const props = {
    activeTab: INFO_HEADER_TABS_ENUM.HOW_TO_USE_THIS,
    onClose: jest.fn(),
    isExpanded: true,
  };
  const tree = renderer.create(<Drawer {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Drawer - collapsed", () => {
  const props = {
    activeTab: INFO_HEADER_TABS_ENUM.HOW_TO_USE_THIS,
    onClose: jest.fn(),
    isExpanded: false,
  };
  const tree = renderer.create(<Drawer {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
