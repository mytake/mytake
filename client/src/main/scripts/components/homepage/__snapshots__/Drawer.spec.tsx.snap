// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer - collapsed 1`] = `
Array [
  <div
    className="drawer"
    style={
      Object {
        "maxHeight": 0,
      }
    }
  >
    <div
      className="drawer__scroller"
    >
      <div
        className="drawer__contents"
      >
        <DrawerContents
          activeTab="What is this"
        />
      </div>
    </div>
  </div>,
  <div
    className="drawer__overlay"
    onClick={[MockFunction]}
  />,
]
`;

exports[`Drawer - expanded 1`] = `
Array [
  <div
    className="drawer drawer--visible"
    style={
      Object {
        "maxHeight": 652.8,
      }
    }
  >
    <div
      className="drawer__scroller"
    >
      <div
        className="drawer__contents"
      >
        <DrawerContents
          activeTab="What is this"
        />
      </div>
    </div>
  </div>,
  <div
    className="drawer__overlay drawer__overlay--visible"
    onClick={[MockFunction]}
  />,
]
`;
